### ISSUE TITLE
Issues triggering Flask routes as scheduled events

### ISSUE THREAD

**Issue by peteryin21** [Miserlou/Zappa#1603]
## Context
Hi! I have a Flask app running with Zappa and I'm having trouble getting the events to trigger the API endpoints as expected.

 I'd like to run some of the functions in my routes.py on a schedule. The closest thing I found to this was the scheduled events with kwargs, so I created the following in my `zappa_settings.json`
```
{
    "dev": {
        "app_function": "grablunch.app",
        "aws_region": "us-west-2",
        "profile_name": "default",
        "project_name": "grablunch",
        "runtime": "python3.6",
        "s3_bucket": "test",
        "memory_size": 3008, // Lambda function memory in MB. Default 512.
        "apigateway_enabled": true, // Set to false if you don't want to create an API Gateway resource. Default true.
        "keep_warm_expression": "rate(44 minutes)", // How often to execute the keep-warm, in cron and rate format. Default 4 minutes.
        "domain": "test.net",
        "slim_handler": true,
        "timeout_seconds": 300,
        "events": [
            {
                "function": "preview_matches", 
                "kwargs": {
                    "day": "thursday"
                },
                "expression": "cron(* * ? * * *)"
            }
        ],
    }
}
```
In my `routes.py` (which is imported at the bottom of my `apps/__init__.py`)
```
@app.route("/some_url/<day>)
def preview_matches(day):
    # stuff
```
I see the function trigger every minute in the cloudwatch logs, with the following Zappa event
```
[DEBUG]	2018-09-07T23:21:08.488Z	b341fac0-b2f4-11e8-a093-e1d21d73af58	Zappa Event: {'time': '2018-09-07T23:21:00Z', 'detail-type': 'Scheduled Event', 'source': 'aws.events', 'account': '703076520410', 'region': 'us-west-2', 'detail': {}, 'version': '0', 'resources': ['arn:aws:events:us-west-2:703076520410:rule/grablunch-dev-send_match_email'], 'id': '27e03715-580f-973f-34da-11ccc2891401', 'kwargs': {'day': 'thursday'}}
``` 
But the function is not actually called. Any ideas?

## Environment

* Zappa version used: 0.46.2
* Operating System and Python version: MacOS Sierra, Python 3.6.4

**Comment by paulclarkaranz** [comment]
I haven't used this feature but it looks like it only accepts function names like `module.function_name` https://github.com/Miserlou/Zappa/blob/12ac82e71deeffcb03dbb6549be4f6f60d537382/zappa/handler.py#L359 and you might need a special scheduled event handling function that pulls the kwargs out of the event (which is the object in the log message you posted) https://github.com/Miserlou/Zappa/blob/12ac82e71deeffcb03dbb6549be4f6f60d537382/zappa/handler.py#L262

**Comment by peteryin21** [comment]
Thanks for the tips @paulclarkaranz. Once I changed the function to the correct path: `app.routes.preview_matches`, the function was called, but I was getting a ```AttributeError: 'NoneType' object has no attribute 'app'```. This error is because I'm trying to call a route without the app context, which Flask doesn't like. 

To workaround it I created a new file called `cloudwatch_methods.py`, made a function
```
def cloudwatch_send_monday_match_email(zappa_event):
    with app.app_context():
        send_match_emails("monday")
```
I originally got the day from the `kwargs` using 
```
day = zappa_event["kwargs"]["day"]
```
but then both methods would have same name and show up as one rule in Cloudwatch, which doesn't work for me since I need two different schedules for them.
